sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -249,11 +241,12 @@ case class Divide(left: Expression, right: Expression)
       s"${eval2.value} == 0"
     }
     val javaType = ctx.javaType(dataType)
-    val divide = if (dataType.isInstanceOf[DecimalType]) {
+    val divide = if (dataType.isInstanceOf[DecimalType] || dataType.isInstanceOf[DoubleType]) {
       s"${eval1.value}.$decimalMethod(${eval2.value})"
     } else {
-      s"($javaType)(${eval1.value} $symbol ${eval2.value})"
+      s"($javaType)(${eval1.value} $decimalMethod ${eval2.value})"
```

Comment:
ok I see the problem. Since `symbol` is also used in the string format of arithmetic expression, `IntegerDivide` has to override it.

I think we can:

1. inline the operator in generated code, e.g. `($javaType)(${eval1.value} / ${eval2.value})` instead of `($javaType)(${eval1.value} $symbol ${eval2.value})`
2. don't override `decimalMethod` as it's useless in the 2 division expressions.